Rewrite HTTP security BBEs (#3791)
ballerina-platform/ballerina-distribution: examples/http-service-basic-authentication-file-user-store/http_service_basic_authentication_file_user_store.md

# HTTP service - Basic authentication file user store

> An `http:Service` can be secured with basic authentication and optionally by enforcing authorization. The service validates the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.

Author: ```suggestion
An `http:Service` can be secured with basic authentication and optionally by enforcing authorization. This can be achieved by validating the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.
```

> An `http:Service` can be secured with basic authentication and optionally by enforcing authorization. The service validates the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on the `user stores`. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.

Author: ```suggestion
An `http:Service` can be secured with basic authentication and optionally by enforcing authorization. The service validates the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.
```

> An `http:Service` can be secured with basic authentication and optionally by enforcing authorization. This can be achieved by validating the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.

Author: ```suggestion
The `http:Service` can be secured with basic authentication and optionally by enforcing authorization. This can be achieved by validating the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.
```

> The `http:Service` can be secured with basic authentication and optionally by enforcing authorization. This can be achieved by validating the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.

Reviewer: ```suggestion
The `http:Service` can be secured with basic authentication and optionally, by enforcing authorization. This can be achieved by validating the basic authentication token sent in the `Authorization` header against the provided configurations in the `Config.toml` file. This file stores the usernames and passwords for the authentication and the scopes for the authorization. An `http:Service` can configure the scopes it needs for authorization in the `auth` field of the `@http:ServiceConfig` annotation. 

A `401 Unauthorized` response is sent to the client when the authentication fails, and a `403 Forbidden` response is sent to the client when the authorization fails. Use this to authenticate and authorize requests based on user stores. Furthermore, the authentication and authorization configurations can be overwritten at the resource level using the `@http:ResourceConfig` annotation.
```
This paragraph is too lengthy. Better to break into two.
